pkg/controller/rolloutblock/rolloutblock_controller.go

```diff
@@ -0,0 +1,622 @@
+			err = c.addApplicationToRolloutBlockStatus(overridingKey, key)
+		}
+	}
+
+	return err
+}
+
+func (c *Controller) syncNewRolloutBlockObject(key string) error {
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	rolloutBlock, err := c.rolloutBlockLister.RolloutBlocks(ns).Get(name)
+	if err != nil {
+		return err
+	}
+
+	if ns == shipper.ShipperNamespace {
+		apps, err := c.applicationLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+		if err = c.addMultipleApplicationsToRolloutBlocks(apps, key, rolloutBlock); err != nil {
+			return err
+		}
+
+		rels, err := c.releaseLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+		if err = c.addMultipleReleasesToRolloutBlocks(rels, key, rolloutBlock); err != nil {
+			return err
+		}
+	} else {
+		apps, err := c.applicationLister.Applications(ns).List(labels.Everything())
+		if err != nil {
+			return err
+		}
+		if err = c.addMultipleApplicationsToRolloutBlocks(apps, key, rolloutBlock); err != nil {
+			return err
+		}
+
+		rels, err := c.releaseLister.Releases(ns).List(labels.Everything())
+		if err != nil {
+			return err
+		}
+		if err = c.addMultipleReleasesToRolloutBlocks(rels, key, rolloutBlock); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Controller) addMultipleReleasesToRolloutBlocks(releases []*shipper.Release, rolloutBlockKey string, rolloutBlock *shipper.RolloutBlock) error {
+	var relsStrings []string
+	for _, release := range releases {
+		if release.DeletionTimestamp != nil {
+			continue
+		}
+
+		relKey, err := cache.MetaNamespaceKeyFunc(release)
+		if err != nil {
+			runtime.HandleError(err)
+			continue
+		}
+
+		overrideRB, ok := release.GetAnnotations()[shipper.RolloutBlocksOverrideAnnotation]
+		if !ok {
+			continue
+		}
+
+		overrideRBs := strings.Split(overrideRB, ",")
+		for _, rbKey := range overrideRBs {
+			if rbKey == rolloutBlockKey {
+				relsStrings = append(relsStrings, relKey)
+			}
+		}
+	}
+
+	if len(relsStrings) == 0 {
+		relsStrings = []string{}
+	}
+
+	rolloutBlock.Status.Overrides.Release = relsStrings
+	_, err := c.shipperClientset.ShipperV1alpha1().RolloutBlocks(rolloutBlock.Namespace).Update(rolloutBlock)
+	if err != nil {
+		return shippererrors.NewKubeclientUpdateError(rolloutBlock, err).
+			WithShipperKind("RolloutBlock")
+	}
+
+	return nil
+}
+
+func (c *Controller) addMultipleApplicationsToRolloutBlocks(applications []*shipper.Application, rolloutBlockKey string, rolloutBlock *shipper.RolloutBlock) error {
+	var appsStrings []string
+	for _, app := range applications {
+		if app.DeletionTimestamp != nil {
+			continue
+		}
+
+		appKey, err := cache.MetaNamespaceKeyFunc(app)
+		if err != nil {
+			runtime.HandleError(err)
+			continue
+		}
+
+		overrideRB, ok := app.GetAnnotations()[shipper.RolloutBlocksOverrideAnnotation]
+		if !ok {
+			continue
+		}
+
+		overrideRBs := strings.Split(overrideRB, ",")
+		for _, rbKey := range overrideRBs {
```

Comment:
I believe this block needs a comment: what kind of override patterns we expect (don't hesitate to drop an example) and what's the expected result.